### DISCUSSION TITLE
pro microがソケット挿入時にショート（？）する

### DISCUSSION THREAD

**Discussion by ghost** [Yowkees/keyball#337] [Q&A] [answered]
pro micro 単体をPCに接続すると 
　pro micro 上の赤いLEDが点灯し、テストファームウェアの書き込みができた。
ピンソケットにpro microを差すと
　pro micro 上の赤いLEDが点灯せず、PCに繋がらない

ちまちま確認してみて、ボールアリ側左側ピンソケット上部に問題がありそうだが
いくらはんだを直しても状況が改善されない。

ボールなし側は問題なく作動した。

ピンソケットを交換することで改善される問題なのでしょうか？

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。

ピンソケットもしくはピンヘッダーの半田付け時に半田を加えすぎると内部に流れ込みショートしてしまうことがあります。
半田を吸い取って除去するのは難しいので、ピンソケットを取り外す必要があります。

除去するときにピンソケットが壊れてしまった場合、Keyballキット購入者向けに部品単位でも販売可能ですので白銀ラボウェブショップのお問い合わせよりご連絡いただければ幸いです。

以上、お手数ですがご確認お願いいたします。

**Comment by ghost** [comment]
ご回答いただきありがとうございます。
ご指摘の通りピンソケットを見直したことで、問題の箇所は改善されました。

本件と関係ない質問の追加で申し訳ないのですが、
左右間の通信がうまくいっておらず
　親側が操作出来て、繋いだ子側がLEDはつくもののまったく操作できないという状況です。
こちらもピンソケットが問題になるのでしょうか。
調べた感じだと互換機で起こる事例というのもあったのですが。

**Comment by ghost** [comment]
他の方のQAを参考に全て解決いたしました！
素晴らしい製品と、サービスをありがとうございます！

**Comment by Yowkees** [comment]
半田付けの修正と、LEDについてもご自身で解決していただき、誠にありがとうございました。
ぜひ末永くご愛用いただければ幸いです。

以上、今後ともどうぞよろしくお願いいたします。
